tests/integration_tests/functional/test_balloon.py

```diff
@@ -14,18 +14,41 @@
 from framework.artifacts import ArtifactCollection, ArtifactSet
 from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
 from framework.matrix import TestMatrix, TestContext
-from framework.utils import get_free_mem_ssh
+from framework.utils import get_free_mem_ssh, run_cmd
 
 import host_tools.network as net_tools  # pylint: disable=import-error
 
 
 MB_TO_PAGES = 256
 
 
-def get_rss_mem_by_pid(pid):
-    """Get the RSS memory that a guest uses, given the pid of the guest."""
-    output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
-    return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+def get_stable_rss_mem_by_pid(pid, percentage_delta=0.5, timeout_secs=30):
+    """
+    Get the RSS memory that a guest uses, given the pid of the guest.
+
+    Wait till the fluctuations in RSS drop below percentage_delta. If timeout
+    is reached before the fluctuations drop, raise an exception.
+    """
+
+    def get_rss_from_pmap():
+        _, output, _ = run_cmd("pmap -X {}".format(pid))
+        return int(output.split('\n')[-2].split()[1], 10)
+
+    rss = get_rss_from_pmap()
+    delta = 100
+    while delta > percentage_delta:
+        prev_rss = rss
+        time.sleep(1)
+
+        rss = get_rss_from_pmap()
+        timeout_secs -= 1
+        if timeout_secs == 0:
+            raise Exception(
+                "Timed out while reading RSS, it did not stabilize")
+
+        delta = (abs(prev_rss - rss)/float(prev_rss)) * 100
+
+    return rss
```

Comment:
You can replace the while with a [retry decorator](https://pypi.org/project/retry/).
```suggestion
@retry(delay=0.5, tries=10)
def get_stable_rss_mem_by_pid(pid, percentage_delta=0.5):
    """
    Get the RSS memory that a guest uses, given the pid of the guest.

    Wait till the fluctuations in RSS drop below percentage_delta. If timeout
    is reached before the fluctuations drop, raise an exception.
    """

    def get_rss_from_pmap():
        _, output, _ = run_cmd("pmap -X {}".format(pid))
        return int(output.split('\n')[-2].split()[1], 10)

    first_rss = get_rss_from_pmap()
    time.sleep(1)
    second_rss = get_rss_from_pmap()
    delta = (abs(first_rss - second_rss)/float(first_rss)) * 100
    assert delta < percentage_delta
    return second_rss
```

and import the retry framework with "from retry import retry".

Reply:
Nice suggestion, updated the PR